cmd/agent/app/options/options.go

```diff
@@ -212,3 +212,12 @@ func NewGrpcProxyAgentOptions() *GrpcProxyAgentOptions {
 	}
 	return &o
 }
+
+func defaultAgentID() string {
+	// Default to the value set by the PROXY_AGENT_ID environment variable. If both the flag &
+	// environment variable are set, the flag always wins.
+	if id := os.Getenv("PROXY_AGENT_ID"); id != "" {
```

Comment:
Not that your trying but its worth noting that we cannot detect multiple agents using the same agent id. We can see that we get repeat connection handshakes from the same agent but that's expected.